src/ng-option-highlight/lib/ng-option-highlight.directive.ts

```diff
@@ -1,57 +1,47 @@
-import { AfterViewInit, Directive, ElementRef, Input, OnChanges, Renderer2 } from '@angular/core';
+import { Directive, ElementRef, input, inject, computed } from '@angular/core';
+
+function escapeRegExp(str: string): string {
+	return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
 
 @Directive({
 	selector: '[ngOptionHighlight]',
 	standalone: true,
+	host: {
+		'innerHTML': 'innerHTML()'
+	}
 })
-export class NgOptionHighlightDirective implements OnChanges, AfterViewInit {
-	@Input('ngOptionHighlight') term: string;
+export class NgOptionHighlightDirective {
+	private readonly elementRef = inject(ElementRef);
 
-	private element: HTMLElement;
-	private label: string;
+	readonly term = input<string>(undefined, { alias: "ngOptionHighlight" });
 
-	constructor(
-		private elementRef: ElementRef,
-		private renderer: Renderer2,
-	) {
-		this.element = this.elementRef.nativeElement;
-	}
+	private element = this.elementRef.nativeElement;;
+	private label: string;
 
 	private get _canHighlight() {
-		return this._isDefined(this.term) && this._isDefined(this.label);
+		return this._isDefined(this.term()) && this._isDefined(this.label);
```

Comment:
Better use `linkedSignal` here or `computed`

Reply:
Updated to use computed - https://github.com/ng-select/ng-select/pull/2584/commits/b6bdb8f10f571578de699adb937c94b54ce5761c#diff-d9fd1e2e487941eb7caa372db5de046840dfe57b409b435c906f946a2f583279R28